decidim-assemblies/app/cells/decidim/assemblies/assembly_metadata_g_cell.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Assemblies
+    # This cell renders the assembly metadata for g card
+    class AssemblyMetadataGCell < Decidim::CardMetadataCell
+      private
+
+      def items
+        [type_item, children_item].compact
+      end
+
+      def type_item
+        return unless has_meta_scope?
+
+        {
+          text: translated_attribute(model.meta_scope),
+          icon: "group-2-line"
+        }
+      end
+
+      def children_item
+        return if children_assemblies_count_for_user.zero?
+
+        {
+          text: t("children_item", count: children_assemblies_count_for_user, scope: "layouts.decidim.assemblies.metadata"),
+          icon: "bubble-chart-line"
+        }
+      end
+
+      def children_assemblies_count_for_user
```

Comment:
Based on the investigation, you do not need the complexity of this method. 
`Assembly` is using `include Decidim::HasPrivateUsers`, which adds the method `visible_for()`

with the current conditions: 

```
      def self.visible_for(user)
        if user 
          return all if user.admin?

          where(
            id: public_spaces +
                private_spaces
                  .joins(:participatory_space_private_users)
                  .where(decidim_participatory_space_private_users: { decidim_user_id: user.id })
          )
        else
          public_spaces
        end
      end
```

As you can see, those are the same statements (admin statement): 
```
a = Decidim::Assembly.first.children.published.to_sql
b = Decidim::Assembly.first.children.published.visible_for(Decidim::User.first).to_sql
a == b
=> true

```
Guest visitor ;
```
a = Decidim::Assembly.first.children.published.public_spaces.to_sql
b = Decidim::Assembly.first.children.published.visible_for(nil).to_sql
a == b
=> true
```

In conclusion, this method should look like: 
```
def children_assemblies_count_for_user
  @children_assemblies_count_for_user ||= published_children_assemblies.visible_for(current_user).count
end 
```

Reply:
Thanks, much better this way